apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/CommonUtils.cs

```diff
@@ -37,6 +40,57 @@ internal static class CommonUtils
                 .Assembly
                 .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
                 .InformationalVersion;
+
+        /// <summary>
+        /// Determines the correct project id from a string project id and a <see cref="MonitoredResource"/>.
+        /// <list type="bullet">
+        /// <term>
+        /// If the string project id is not null and the MonitoredResource does not contain a project id
+        /// then the string project id is retruned.
+        /// </term>
+        /// <term>
+        /// If the string project id is not null and the MonitoredResource contains a project id
+        /// then the string project id is retruned.  We do this as the explictly set project id should be
+        /// defaulted too.
+        /// </term>
+        /// <term>
+        /// If the string project id is null and the MonitoredResource does contains a project id
+        /// then the project id from the MonitoredResource is retruned.
+        /// </term>
+        /// <term>
+        /// If the string project id is null and the MonitoredResource does not contain a project id
+        /// then an <see cref="InvalidOperationException"/> is thrown.
+        /// </term>
+        /// </list>
```

Comment:
This bulleted list is a bit verbose. Perhaps this will suffice instead: "If the specified project id is not null, it is returned. Otherwise, if the project id of the MonitoredResource is not null, it is returned. It both are null, an \<see cref="InvalidOperationException"/\> is thrown."